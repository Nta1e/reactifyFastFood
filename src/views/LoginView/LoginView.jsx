import React, {Component} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { toast } from 'react-toastify';
import Login from '../../components/LoginComponent/Login';
import loginAction from '../../actions/loginActions'


export class LoginView extends Component {
    state = {};
    componentWillReceiveProps(nextProps){
        const {error, data} = nextProps;
        if(error.length > 0){
            toast.error(error);
            setTimeout(()=>{
                window.location.reload();
            }, 3800)
        }else if(data.message.length > 0){
            toast.success(data.message);
            window.localStorage.setItem('token', data.token)
            setTimeout(()=>{
                window.location.href='/'
            }, 3000)

        }
    }
    onChange = (e) => {
        this.setState({[e.target.name]: e.target.value})
    };
    onSubmit = (e) => {
        e.preventDefault();
        const {loginAction} = this.props;
        const postData = {
            ...this.state
        };
        loginAction(postData)
    };
    render() {
        return (
          <Login onChange={this.onChange} onSubmit={this.onSubmit} />
        );
    }
}

LoginView.propTypes = {
    loginAction: PropTypes.func.isRequired,
    history: PropTypes.func.isRequired,
    error: PropTypes.string.isRequired,
};

export const mapStateToProps = state => ({
    error: state.login.error,
    data: state.login.success
});
export default connect(
    mapStateToProps,
    {loginAction}
)(LoginView);


