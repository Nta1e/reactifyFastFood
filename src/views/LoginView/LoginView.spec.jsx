import React from 'react';
import { shallow } from 'enzyme';
import { LoginView, mapStateToProps } from './LoginView';

const loginAction = jest.fn();
const push = jest.fn();
const props = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
    loginAction,
    history: {push}
};
const state = {
    login: {
        error: 'error',
        success: ''
    }
};

describe('test login view', ()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = shallow(<LoginView {...props} />)
    })
    it('matches snapShot',()=>{
        expect(wrapper).toMatchSnapshot()
    });
    it('maps state to props',()=>{
        expect(mapStateToProps(state).error).toEqual('error')
    });
    it('test on change', ()=>{
        wrapper.instance().onChange({target: {name: 'username', value: 'ntale'}});
        expect(wrapper.state('username')).toEqual('ntale')
    });
    it('test on submit', ()=>{
        wrapper.instance().onSubmit({preventDefault: jest.fn()})
        expect(loginAction).toBeCalled()
    });
    it('test will receive props',()=>{
        wrapper.setProps({error: 'error', success: 'success'})
    })
});