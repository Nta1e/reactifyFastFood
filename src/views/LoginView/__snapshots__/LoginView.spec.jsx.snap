// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login view matches snapShot 1`] = `ShallowWrapper {}`;
